tests/test_skvbc_fast_path.py

```diff
@@ -0,0 +1,158 @@
+            config = bft.TestConfig(n=bft_config['n'],
+                                    f=bft_config['f'],
+                                    c=bft_config['c'],
+                                    num_clients=bft_config['num_clients'],
+                                    key_file_prefix=KEY_FILE_PREFIX,
+                                    start_replica_cmd=start_replica_cmd)
+            with bft.BftTestNetwork(config) as bft_network:
+                await bft_network.init()
+                bft_network.start_all_replicas()
+
+                for _ in range(10):
+                    key, val = await bft_network.write_known_kv(self.protocol)
+
+                await bft_network.assert_fast_path_prevalent(as_of_seq_num=1)
+
+                await bft_network.assert_kv_write_executed(self.protocol, key, val)
+
+    def test_fast_to_slow_path_transition(self):
+        """
+        This test aims to check the correct transition from fast to slow commit path.
+
+        First we write a series of known K/V entries, making sure
+        we stay on the fast path.
+
+        Once the first series of K/V writes have been processed, we bring down
+        one of the replicas, which should trigger a transition to the slow path.
+
+        We send a new series of K/V writes and make sure they
+        have been processed using the slow commit path.
+
+        Finally we check if a known K/V has been executed.
+        """
+        trio.run(self._test_fast_to_slow_path_transition)
+
+    async def _test_fast_to_slow_path_transition(self):
+        for bft_config in bft.interesting_configs():
+            config = bft.TestConfig(n=bft_config['n'],
+                                    f=bft_config['f'],
+                                    c=bft_config['c'],
+                                    num_clients=bft_config['num_clients'],
+                                    key_file_prefix=KEY_FILE_PREFIX,
+                                    start_replica_cmd=start_replica_cmd)
+            with bft.BftTestNetwork(config) as bft_network:
+                await bft_network.init()
+                bft_network.start_all_replicas()
+
+                for _ in range(10):
+                    await bft_network.write_known_kv(self.protocol)
+
+                await bft_network.assert_fast_path_prevalent(as_of_seq_num=1)
+
+                unstable_replicas = list(set(range(0, config.n)) - {0})
+                bft_network.stop_replica(
+                    replica=random.choice(unstable_replicas))
+
+                for _ in range(10):
+                    key, val = await bft_network.write_known_kv(self.protocol)
+
+                await bft_network.assert_slow_path_prevalent(as_of_seq_num=10)
+
+                await bft_network.assert_kv_write_executed(self.protocol, key, val)
+
+    @unittest.skip("According to the SBFT spec, the fast path should tolerate up to c crashed nodes."
+                   "Enable the test once this resilience gets implemented.")
```

Comment:
As discussed offline with @salieri11, we have three commit path modes: 
- optimistic fast path (all 3f + 2c + 1 nodes are available)
- threshold fast path (between 2f + c + 1 and 3f + 2c + 1 nodes are available)
- slow path (no more than 2f + c + 1 nodes are available)

Whenever a replica crashes, we automatically degrade to slow path, but after a timeout expires, we return to threshold fast path (see ControllerWithSimpleHistory).